tests/common.py

```diff
@@ -145,18 +146,58 @@ def post_model_outputs_and_maybe_allow_retries(
     stub: service_pb2_grpc.V2Stub,
     request: service_pb2.PostModelOutputsRequest,
     metadata: Tuple,
+    retryable_output_codes: List[int] = [],
 ):
-    return _retry_on_504_on_non_prod(lambda: stub.PostModelOutputs(request, metadata=metadata))
+    response = _retry_on_504_on_non_prod(stub.PostModelOutputs, request=request, metadata=metadata)
+    # this inspect the individual outputs to see if the code means we should retry prediction (returns if retryable_output_codes not passed)
+    response = _retry_on_unsuccessful_predicts_on_non_prod(
```

Comment:
This assumes that we can't have mixed reasons amongst retried inferences since we only retry on 504 in the first method, and only retry on failure in the second method.  These should be merged into a single retry func.

Reply:
Ah ok, maybe leave a comment about this essentially being a warm up function and/or maybe rename it.